### ISSUE TITLE
RuntimeError: cannot reuse already awaited coroutine

### ISSUE THREAD

**Issue by starflows** [ronf/asyncssh#702]
The code below aborts with `RuntimeError: cannot reuse already awaited coroutine` as soon as the process ends.

To reproduce:

1. run the code below
2. in the "remote" terminal type `exit`

```python
import os
import sys
import asyncio
import asyncssh

async def main():
    async with asyncssh.connect('localhost') as conn:
        async with conn.create_process(
            stdin=os.dup(sys.stdin.fileno()),
            stdout=os.dup(sys.stdout.fileno()),
            stderr=os.dup(sys.stderr.fileno()),
            term_type=os.environ.get('TERM'),
        ) as process:
            await process.wait()

asyncio.run(main())
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "<stdin>", line 3, in main
  File "/home/test/.venv/lib/python3.12/site-packages/asyncssh/misc.py", line 308, in __aexit__
    exit_result = await self._coro_result.__aexit__(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/test/.venv/lib/python3.12/site-packages/asyncssh/process.py", line 831, in __aexit__
    await self.wait_closed()
  File "/home/test/.venv/lib/python3.12/site-packages/asyncssh/process.py", line 1212, in wait_closed
    await task
RuntimeError: cannot reuse already awaited coroutine
```

The code does _not_ raise a `RuntimeError` with asyncssh version 2.14.2
It does raise the error with asyncssh version 2.17.0 as well as with the develop and master branch.

Maybe there is another way to connect your local terminal to a remote process?

Environment:
asyncssh 2.17.0
python 3.12.6

**Comment by starflows** [comment]
Thank you for your reply @ronf !

The workaround and the patch both work and solved my issue.

**Comment by ronf** [comment]
Thanks for the confirmation -- this is now in the "develop" branch as commit bfa04aa.

**Comment by ronf** [comment]
This fix is now available in AsyncSSH 2.18.0.
